popup/js/popup.js

```diff
@@ -209,9 +221,27 @@ function themeChange() {
 
 }
 
+function recordSearchHistory()
+{
+
+        var saveQuery=[];
+        if(localStorage.getItem('search'))
+        saveQuery=JSON.parse(localStorage.getItem('search'));
+        var x=text.value;
+        if(x!="" && saveQuery.indexOf(x)==-1)
+        {
+        saveQuery.push(x);
+        localStorage.setItem('search',JSON.stringify(saveQuery));
+        }
+}
+
 
 document.addEventListener("DOMContentLoaded", function () {
-    document.querySelector("button").addEventListener("click", register);
+    document.querySelector("button").addEventListener("click",function()
```

Comment:
We need a style linter.

Reply:
Could you please elaborate? I don't know what a linter is

Reply:
Oh sure. Linters help to find potential errors in code. You can look into eslint and prettier(style linter - not necessarily errors but helps to have standard coding style throughout the project).

Reply:
I have corrected all the errors in the updated files in accordance with the Airbnb configuration of ESlint that was already setup. I have also added a max-len rule in the .eslintrc file.

Reply:
I hope you had fun learning about linters they have a pretty integral part in development.

I apologies so many changes might seem intimidating. But it is happening because this is a really big feature and is introducing a lot code into the codebase. Imagine if we straightaway merge this code, people who might want to contribute next year will face so much trouble to add more awesome features like these. Hope you understand.

Reply:
Of course. This experience is teaching me a lot that I would probably haven't learnt otherwise